lib/elixir/lib/kernel.ex

```diff
@@ -833,6 +837,17 @@ defmodule Kernel do
       tl([])
       #=> ** (ArgumentError) argument error
 
+      tl([:one])
+      #=> []
+
+  Given an improper list, it returns an improper list or the its last element.
```

Comment:
I don't think we need this clarification. The tail of an improper list is an improper list or the last element. The examples are welcome but those are properties and when listed in the documentation as so they feel like special cases.